app-backend/api/src/main/scala/annotation-project/AnnotationProjectTaskRoutes.scala

```diff
@@ -354,16 +354,90 @@ trait AnnotationProjectTaskRoutes
         ScopedAction(Domain.AnnotationProjects, Action.CreateAnnotation, None),
         user
       ) {
-        authorizeAuthResultAsync {
-          AnnotationProjectDao
-            .authorized(
-              user,
-              ObjectType.AnnotationProject,
-              projectId,
-              ActionType.Annotate
+        authorizeAsync {
+          (for {
+            auth1 <- AnnotationProjectDao
+              .authorized(
+                user,
+                ObjectType.AnnotationProject,
+                projectId,
+                ActionType.Annotate
+              )
+            auth2 <- AnnotationProjectDao
+              .authorized(
+                user,
+                ObjectType.AnnotationProject,
+                projectId,
+                ActionType.Validate
+              )
+            auth3 <- TaskDao.hasStatus(
+              taskId,
+              List(
+                TaskStatus.Unlabeled,
+                TaskStatus.LabelingInProgress,
+                TaskStatus.Labeled
+              )
             )
-            .transact(xa)
-            .unsafeToFuture
+          } yield {
+            (auth1.toBoolean || auth2.toBoolean) && auth3
+          }).transact(xa).unsafeToFuture
+        } {
+          entity(as[AnnotationLabelWithClassesFeatureCollectionCreate]) { fc =>
+            val annotationLabelWithClassesCreate = fc.features map {
```

Comment:
This block seems a bit repetitive, like it is the same as the implementation that follows this endpoint. I am wondering if we could define a private function that does the thing and call it in both spots -- similar to how we dealt with the lock and unlock endpoints.